src/main/java/net/mcreator/ui/dialogs/NewVariableDialog.java

```diff
@@ -66,9 +68,10 @@ public static VariableElement showNewVariableDialog(JFrame frame, boolean showSc
 				&& textField.getValidationStatus().getValidationResultType() != Validator.ValidationResultType.ERROR
 				&& type.getSelectedItem() != null) {
 			VariableElement element = new VariableElement();
+			VariableElementType variable = VariableElementTypeLoader.getVariableTypeFromString(((VariableElementType) type.getSelectedItem()).getName());
 			element.setName(Transliteration.transliterateString(textField.getText()));
 			element.setType((VariableElementType) type.getSelectedItem());
-			element.setValue(VariableElement.getDefaultValueForType((VariableElementType) type.getSelectedItem()));
+			element.setValue(Objects.requireNonNull(variable).getDefaultValue());

```

Comment:
I would check if variable is null and in this case return null from this method (as if frame is closed)